pkg/virt-controller/watch/vmi_test.go

```diff
@@ -573,6 +588,28 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			shouldExpectVirtualMachineDataVolumesReadyCondition(vmi, k8sv1.ConditionFalse)
 			controller.Execute()
 		})
+
+		It("should not create a corresponding Pod on VMI creation when backend storage is pending", func() {
+			vmi := NewPendingVirtualMachine("testvmi")
+			vmi.Spec.Domain.Firmware = &virtv1.Firmware{
+				Bootloader: &virtv1.Bootloader{
+					EFI: &virtv1.EFI{
+						Persistent: pointer.Bool(true),
+					},
+				},
+			}
+
+			pvc := NewPvc(vmi.Namespace, backendstorage.PVCForVMI(vmi))
+			pvc.Status.Phase = k8sv1.ClaimPending
+			Expect(pvcInformer.GetIndexer().Add(pvc)).To(Succeed())
+
+			addVirtualMachine(vmi)
+			kubeClient.Fake.PrependReactor("get", "persistentvolumeclaims", func(action testing.Action) (handled bool, obj k8sruntime.Object, err error) {
+				return true, pvc, nil
+			})
+			shouldExpectVirtualMachinePendingState(vmi)
+			controller.Execute()
+		})
```

Comment:
If you add a unit test to create pod in Pending+WFFC case I will be good approving this PR with no dunctest

Reply:
Done, PTAL!
/hold cancel